cpp/include/raft/core/mdspan.hpp

```diff
@@ -304,4 +304,24 @@ RAFT_INLINE_FUNCTION auto unravel_index(Idx idx,
   }
 }
 
+/**
+ * @brief Create a copy of the given mdspan with const element type
+ * @tparam mdspan_type Expected type raft::host_mdspan or raft::device_mdspan
+ * @param mds raft::host_mdspan or raft::device_mdspan object
+ * @return raft::host_mdspan or raft::device_mdspan with vector_extent
+ *         depending on AccessoryPolicy
+ */
+template <typename mdspan_type, typename = enable_if_mdspan<mdspan_type>>
+auto make_const_mdspan(mdspan_type mds)
```

Comment:
This is great! Can you add the corresponding `doxygenfunction` to the docs (`docs/cpp_api/mdspan_mdspan.rst`) as well?